sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -104,6 +105,11 @@ def _create_environment(options):
   def run_pipeline(self, pipeline):
     portable_options = pipeline.options.view_as(PortableOptions)
     job_endpoint = portable_options.job_endpoint
+
+    # portable runner specific default
+    if pipeline.options.view_as(SetupOptions).sdk_location == 'default':
+      pipeline.options.view_as(SetupOptions).sdk_location = 'container'
```

Comment:
I wonder, shouldn't we change the interpretation of "default" in case of the portable runner to be `container`, instead of changing the value here?

Reply:
For Dataflow, default means pypi.
Default can mean different thing based on the runner so translating default to container seems reasonable.

Reply:
If we can change the interpretation of `default` in the FlinkRunner to be `container`, I'd prefer that. It looks like changing this here could cause unexpected behavior.

Reply:
It is kind of strange to have that `'default'` default value. Should it not be just `None` when there is no real default? Maybe we can change that in a follow-up PR.

Reply:
@mxm I don't think it can be deferred to the runner, the stager behavior depends on it. I would prefer that the stager does nothing unless a specific option was set (like `pypi` in dataflow case), but that should probably be a separate change. It would allow us to not even touch the value in portable runner.

Reply:
You're right, that doesn't work. I just find it odd to rewrite `default` -> `container` when the default could be `container`. The problem is that this option is used in different contexts (Portable Runner / Dataflow Runner).

Couldn't we alter the behavior of `stager` when the default value is used?

Reply:
https://issues.apache.org/jira/browse/BEAM-5525